core/src/main/java/com/novoda/merlin/RxCallbacksManager.java

```diff
@@ -0,0 +1,25 @@
+package com.novoda.merlin;
+
+import rx.Observable;
+import rx.subjects.BehaviorSubject;
+
+public class RxCallbacksManager {
+
+    private final BehaviorSubject<Boolean> merlinRxConnectionStatusSubject;
```

Comment:
:+1: for extensibility!
On Oct 27, 2015 10:07 AM, "Benjamin AUGUSTIN" <notifications@github.com>
wrote:

> In core/src/main/java/com/novoda/merlin/RxCallbacksManager.java
> <https://github.com/novoda/merlin/pull/72#discussion_r43102141>:
>
> > @@ -0,0 +1,25 @@
> > +package com.novoda.merlin;
> > +
> > +import rx.Observable;
> > +import rx.subjects.BehaviorSubject;
> > +
> > +public class RxCallbacksManager {
> > +
> > +    private final BehaviorSubject<Boolean> merlinRxConnectionStatusSubject;
>
> I would usually agree with you since YAGNI
> <https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it>. In this case
> though since we are building a public API and it is not a big change in
> code I would go for the enum. It will allow for non breaking changes with
> regards to the API.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/merlin/pull/72/files#r43102141>.
>

-- 
Disclaimer: The information in this e-mail and any attachments is the 
property of Novoda Ltd and is confidential and may be legally privileged. 
This e-mail is intended solely for the person or organisation to which it 
is addressed. Any disclosure, copying or other use of the information by 
any person or organisation who is not the intended recipient is strictly 
prohibited and may be unlawful. If you have received this e-mail in error, 
please inform the sender immediately and delete/destroy this e-mail and any 
copies of it. Novoda Ltd has taken reasonable precautions to minimise the 
risk of any software viruses which may damage your systems, but we advise 
that you take the necessary steps to ensure that no virus contamination is 
suffered. Novoda Ltd does not accept any liability for any loss or damage 
caused by the transmission of any virus.

Novoda Ltd, Company No: 347444, Registered in Scotland Registered 
Office: C/O Alexander Sloan, 38 Cadogan Street, Glasgow, G2 7HF, Scotland.  VAT 
Registration Number  GB 984 2525 93